packages/eslint-plugin-pf-codemods/src/rules/v6/emptyStateHeaderMoveIntoEmptyState/emptyStateHeader-move-into-emptyState.ts

```diff
@@ -0,0 +1,261 @@
+        return node.openingElement.name.name === componentName;
+      }
+
+      return false;
+    };
+
+    const getAttribute = (
+      node: JSXElement,
+      attributeName: string
+    ): JSXAttribute | undefined => {
+      const attributes = node.openingElement.attributes.filter(
+        (attr) => attr.type === "JSXAttribute"
+      ) as JSXAttribute[];
+      return attributes.find((attr) => attr.name.name === attributeName);
+    };
+
+    const getExpressionValue = (node?: JSXAttribute["value"]) => {
+      if (!node) {
+        return;
+      }
+
+      if (node.type === "JSXExpressionContainer") {
+        return node.expression;
+      }
+    };
+
+    const getAttributeText = (attribute?: JSXAttribute) => {
+      if (!attribute) {
+        return "";
+      }
+
+      return context.getSourceCode().getText(attribute);
+    };
+
+    const getAttributeValueText = (attribute?: JSXAttribute) => {
+      if (!attribute || !attribute.value) {
+        return "";
+      }
+
+      const { value } = attribute;
+
+      if (value.type === "Literal") {
+        return value.value;
+      }
+
+      return "";
+    };
+
+    const getNodesText = (nodes: Node[]) => {
+      return nodes
+        .map((node) => context.getSourceCode().getText(node))
+        .join("");
+    };
+
+    const getElementChildText = (children: JSXElement["children"]) => {
+      if (!children.length) {
+        return "";
+      }
+
+      switch (children[0].type) {
+        case "JSXText":
+          return (children as JSXText[]).map((child) => child.value).join("");
+        case "JSXExpressionContainer":
+        case "JSXSpreadChild":
+          return getNodesText(
+            children.map(
+              (child) =>
+                (child as JSXExpressionContainer | JSXSpreadChild).expression
+            )
+          );
+        case "JSXElement":
+        case "JSXFragment":
+          return getNodesText(children as JSXElement[] | JSXFragment[]);
+        default:
+          return "";
+      }
+    };
+
+    return {
+      ImportDeclaration(node: ImportDeclaration) {
+        if (
+          !pfPackageMatches(pkg, node.source.value) ||
+          !includesImport(node.specifiers, "EmptyStateHeader")
+        ) {
+          return;
+        }
+      },
+      JSXElement(node: JSXElement) {
+        if (!isComponentNode(node, "EmptyState")) {
+          return;
+        }
+
+        const header = getChildElementByName("EmptyStateHeader", node);
+
+        if (!header || header.type !== "JSXElement") {
+          return;
+        }
+
+        const headingClassNameAttribute = getAttribute(header, "className");
+        const headingLevelAttribute = getAttribute(header, "headingLevel");
+        const titleClassNameAttribute = getAttribute(header, "titleClassName");
+        const titleTextAttribute = getAttribute(header, "titleText");
+        const headerIconAttribute = getAttribute(header, "icon");
```

Comment:
Nit not a blocker: moving some of these and various other variables where they're needed rather than all grouped in a single spot